lib/hanami/utils/kernel.rb

```diff
@@ -404,15 +413,14 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
```

Comment:
@pascalbetz I've replied to that Gist. The difference between switch and if/else is about 10%. You can use `x.compare!` to let `benchmark-ips` to calculate the diff for you :wink: 

If you have time, feel free to open a PR with this speed improvement.